Exiled.API/Features/Room.cs

```diff
@@ -112,49 +112,20 @@ public Room(string name, Transform transform, Vector3 position)
         /// </summary>
         /// <param name="other">Other Room.</param>
         /// <returns>If the rooms are equal.</returns>
-        public bool Equals(Room other)
-        {
-            if (ReferenceEquals(null, other))
-            {
-                return false;
-            }
-
-            if (ReferenceEquals(this, other))
-            {
-                return true;
-            }
-
-            return Id == other.Id;
-        }
+        public bool Equals(Room other) => other is Room room && this == room;
```

Comment:
It'll throw a StackOverflowException.
```suggestion
        public bool Equals(Room other) => other != null && Id == other.Id;
```